### ISSUE TITLE
Checkbox state change

### ISSUE THREAD

**Issue by reactnativedeveloper** [crazycodeboy/react-native-check-box#19]
I am designing filter page like shopping sites where i need multiple check boxes corresponding to different categories. 

Everything is working fine, but when i am de-selecting any checkbox in particular category on particular index then the same change is reflecting on another category.  I just want that whenever i should select any checkbox in particular category the same will not reflect on another category.

Below is the screenshot attached for the same.

I tried lots of solution to the problem but nothing is working for me  ;'(

![screenshot_1495016223](https://cloud.githubusercontent.com/assets/28753533/26149496/1e76f11a-3b18-11e7-8a80-835279093c4f.png)
![screenshot_1495016221](https://cloud.githubusercontent.com/assets/28753533/26149495/1e762456-3b18-11e7-87d2-5bb6151e9366.png)
![screenshot_1495016213](https://cloud.githubusercontent.com/assets/28753533/26149494/1e75bfa2-3b18-11e7-8e56-5760f088300a.png)
![screenshot_1495016208](https://cloud.githubusercontent.com/assets/28753533/26149498/1e7c58d0-3b18-11e7-995a-bb5cce69f248.png)

**Comment by ianhowe76** [comment]
My guess is that when you switch category the checkbox components are not being unmounted but instead they are being re-rendered with a new label

1. Are you using the latest code and setting checked for each checkbox or relying on the internal state of the checkbox?
2. Are you rendering the checkboxes as an array? If so are you setting a unique key such as the label?

**Comment by reactnativedeveloper** [comment]
Yes, i am using latest code. I am using listview  for multiple check boxes and its label.

**Comment by FlaviooLima** [comment]
Having the same problem

**Comment by FlaviooLima** [comment]
Found that I had my view receiving props and refreshing, and when this happens the component start to "mal function".
